gluten-core/src/main/java/io/glutenproject/memory/memtarget/ThrowOnOomMemoryTarget.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.memory.memtarget;
+
+import io.glutenproject.memory.MemoryUsageStats;
+
+import org.apache.spark.memory.SparkMemoryUtil;
+
+public class ThrowOnOomMemoryTarget implements MemoryTarget {
+  private final TaskManagedMemoryTarget target;
+
+  public ThrowOnOomMemoryTarget(TaskManagedMemoryTarget target) {
+    this.target = target;
+  }
+
+  @Override
+  public long borrow(long size) {
+    long granted = target.borrow(size);
+    if (granted < size) {
+      // OOM
+      if (granted != 0L) {
```

Comment:
this check can be removed, right?

Reply:
`target.repay(0)` may lead to unexpected behavior if the underlying target wasn't coded with respect to this case.

Let's keep the check as some sort of defensive practice.